Colw/signup steps

colw opened the pull request:
Closes #2894 

**Description:**

- Use separate module to keep signup form
- Break signup into separate steps

Up for comment before cleaning code.

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

jbibla commented:
looks good. few thoughts: 

- perhaps we could name them by their content as opposed to 1,2,3
- should we enable this stuff in the browser with a warning @faboweb ?
- this should apply to recover as well as sign up

colw commented:
> looks good. few thoughts:
> 
> * perhaps we could name them by their content as opposed to 1,2,3
> * should we enable this stuff in the browser with a warning @faboweb ?
> * this should apply to recover as well as sign up

- I've renamed them now, and cleaned the code.
- FYI, I've enabled the warning just for the build previews. But maybe we could keep it?
- I'll do that once we're happy with this.

faboweb commented:
> should we enable this stuff in the browser with a warning @faboweb ?

Happy to enable in the browser. But let's do in another PR. #2992 2992

faboweb commented on src/components/common/InsecureModeWarning.vue:
```diff
@@ -0,0 +1,28 @@
+<template>
+  <div>
+    <p>
+      Creating an address in the browser is unsafe. To offer you a secure
```
Proposing a new warning (branding + extension is available)

```
Did you know that storing your key in the browser is considered unsafe? We offer a more secure way to storing your keys via our _browser extension_. Check it out in a quiet minute.
```

colw replied:
I like that

faboweb replied:
@mariopino did you see this comment?

mariopino replied:
> @mariopino did you see this comment?

Yes but for some reason i thought it was already done.

What link should we include in ` _browser extension_` link? Chrome store?

mariopino replied:
> > @mariopino did you see this comment?
> 
> Yes but for some reason i thought it was already done.
> 
> @faboweb What link should we include in ` _browser extension_` link? Chrome store?

Autoanswer: http://bit.ly/lunie-ext fits well here

mariopino replied:
![image](https://user-images.githubusercontent.com/9256178/66825287-4c67cc00-ef4a-11e9-8b08-1d9a7cc38dfe.png)

faboweb replied:
![image](https://media.giphy.com/media/xUOwG3MyaiItznFs6k/source.gif)

mariopino commented:
Both `/create` and `/recover` flows working! Lot of testing work pending...

jbibla commented:
awesome work here @colw and @mariopino 

few thoughts about the flow / ux:
- the buttons should say "next" until the final step "create"
- maybe the warning could be displayed _only_ once, as the first step (in the browser)

> A message should go here.

made me laugh 😆

mariopino commented:
> awesome work here @colw and @mariopino
> 
> few thoughts about the flow / ux:
> 
> * the buttons should say "next" until the final step "create"
> * maybe the warning could be displayed _only_ once, as the first step (in the browser)
> 
> > A message should go here.
> 
> made me laugh 😆

I changed the button messages in both flows and display the warning in the first step of create.

Would be nice some inspiration for the success pages btw :-)

jbibla commented:
few thoughts: 

- unless the session form is longer than the screen, it should not scroll
- buttons should only be full width on mobile, tablet and up can be regular sized
- danger screen can be on its own as first step (only in browser)
- recover screen has some obvious styling issues on mobile
- success screen button should say "Go to Lunie" instead of "Back to Lunie"

- have you looked at the impact of these changes on the extension?

jbibla commented:
another thought: 

- let's make the back button a bit smaller and put it in a little circle like this:
<img width="42" alt="Screen Shot 2019-10-03 at 12 55 41 PM" src="https://user-images.githubusercontent.com/6021933/66147624-30c10500-e5dd-11e9-839b-204262b8c971.png">

mariopino commented:
Working in progress:

- Buttons are now small and right aligned by default in all steps, and full width in screens < 667px wide.
- Success screen button now says "Go to Lunie" instead of "Back to Lunie"
- Now back button looks like this:

![image](https://user-images.githubusercontent.com/9256178/66414956-c3590e00-e9fa-11e9-9416-ad6f89dc9dcb.png)

faboweb commented on src/vuex/modules/session.js:
```diff
@@ -15,7 +15,7 @@ export default () => {
     developmentMode: config.development, // can't be set in browser
     experimentalMode: config.development, // development mode, can be set from browser
-    insecureMode: false, // show the local signer
+    insecureMode: true, // show the local signer
```
don't forget to remove this again ^^

faboweb replied:
dangerous to set it in code. better use the query parameter: `?insecure=true`

mariopino replied:
> don't forget to remove this again ^^

Yeah, Colin was farsighted and wrote a test to check that it's false :-)

faboweb replied:
Good guy Colin

faboweb commented on changes/colw_signup-steps:
```diff
@@ -0,0 +1 @@
+[Changed] [#2894](https://github.com/cosmos/lunie/issues/2894) Refactor signin steps into separate screens, including a success screen. @colw
```
```suggestion
[Changed] [#2894](https://github.com/cosmos/lunie/issues/2894) Refactor signin steps into separate screens, including a success screen. @colw @mariopino
```

faboweb commented on src/components/common/DangerZoneWarning.vue:
```diff
@@ -0,0 +1,38 @@
+<template>
```
Nice extraction

faboweb commented on src/initializeApp.js:
```diff
@@ -18,8 +18,11 @@ function setOptions(urlParams, store) {
+  if (config.mobileApp || urlParams.insecure === `true`) {
+    store.commit(`setInsecureMode`, true)
+  }
+  if (urlParams.insecure === `false`) {
```
This is the default, why set it?

mariopino replied:
The flag can be set to true in store and you should also have the possibility to change to false via url param. I'm happy to revert it if you find it's not useful.

faboweb commented on src/routes.js:
```diff
@@ -127,6 +127,57 @@ export default [
       feature: "Session"
     }
   },
+  {
```
The whole routing schema looks excessive. :/ But let's leave it for now.

mariopino replied:
Agree, needed by current approach, but we can get rid off at refactor.

faboweb commented:
Typo: "out" should be "our"

Mario Pino <notifications@github.com> schrieb am Di., 15. Okt. 2019, 12:50:

> *@mariopino* commented on this pull request.
> ------------------------------
>
> In src/components/common/InsecureModeWarning.vue
> <https://github.com/luniehq/lunie/pull/2985#discussion_r334880872>:
>
> > @@ -0,0 +1,28 @@
> +<template>
> +  <div>
> +    <p>
> +      Creating an address in the browser is unsafe. To offer you a secure
>
> [image: image]
> <https://user-images.githubusercontent.com/9256178/66825287-4c67cc00-ef4a-11e9-8b08-1d9a7cc38dfe.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/luniehq/lunie/pull/2985?email_source=notifications&email_token=ABMY5WMC3FJW5GCP2ZKWTKDQOWN65A5CNFSM4IWKNQVKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCH6VPTQ#discussion_r334880872>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABMY5WN274JNQBXEFU3AM33QOWN65ANCNFSM4IWKNQVA>
> .
>

faboweb commented:
In which usecase would you set to insecure=false via the url?

Mario Pino <notifications@github.com> schrieb am Di., 15. Okt. 2019, 12:39:

> *@mariopino* commented on this pull request.
> ------------------------------
>
> In src/initializeApp.js
> <https://github.com/luniehq/lunie/pull/2985#discussion_r334876437>:
>
> > @@ -18,8 +18,11 @@ function setOptions(urlParams, store) {
>    if (urlParams.rpc) {
>      store.commit(`setRpcUrl`, urlParams.rpc)
>    }
> -  if (config.mobileApp || urlParams.insecure) {
> -    store.commit(`setInsecureMode`)
> +  if (config.mobileApp || urlParams.insecure === `true`) {
> +    store.commit(`setInsecureMode`, true)
> +  }
> +  if (urlParams.insecure === `false`) {
>
> The flag can be set to true in store and you should also have the
> possibility to change to false via url param. I'm happy to revert it if you
> find it's not useful.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/luniehq/lunie/pull/2985?email_source=notifications&email_token=ABMY5WNU2K3AMNJHPDRHTELQOWMWFA5CNFSM4IWKNQVKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCH6UFII#discussion_r334876437>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABMY5WK6LON2N6R67TPI2GTQOWMWFANCNFSM4IWKNQVA>
> .
>

jbibla commented on src/components/common/DangerZoneWarning.vue:
```diff
@@ -0,0 +1,38 @@
+  <div class="danger-zone">
+    <h2>DANGER ZONE</h2>
+    <p>
+      Did you know that storing your key in the browser is considered unsafe? We
```
```suggestion
      Did you know that creating your address in the browser is considered unsafe? We
```

jbibla commented on src/components/common/DangerZoneWarning.vue:
```diff
@@ -0,0 +1,38 @@
+    <h2>DANGER ZONE</h2>
+    <p>
+      Did you know that storing your key in the browser is considered unsafe? We
+      offer a more secure way to storing your keys via our
```
```suggestion
      offer a more secure way to do this with our
```

jbibla commented on src/components/common/DangerZoneWarning.vue:
```diff
@@ -0,0 +1,38 @@
+        target="_blank"
+        rel="noopener norefferer"
+        >browser extension</a
+      >. Check it out in a quiet minute. Soon you will also have out mobile app
```
```suggestion
      >. Check it out in a quiet minute.
```

jbibla commented on src/components/common/DangerZoneWarning.vue:
```diff
@@ -0,0 +1,38 @@
+        rel="noopener norefferer"
+        >browser extension</a
+      >. Check it out in a quiet minute. Soon you will also have out mobile app
+      available to store your keys. Stay tuned!
```
```suggestion

```

jbibla commented on src/components/common/TmSessionSignUp.vue:
```diff
@@ -1,249 +1,98 @@
-            type="required"
-          />
-        </TmFormGroup>
+          <DangerZoneWarning />
```
can we put this above line 4?

jbibla commented on src/components/common/TmSessionSignUpSeed.vue:
```diff
@@ -0,0 +1,125 @@
+          <TmFormGroup
+            field-id="sign-up-seed"
+            class="sign-up-seed-group"
+            field-label="Seed phrase"
```
```suggestion
            field-label="Backup Code"
```

jbibla commented on src/components/common/TmSessionImportSuccess.vue:
```diff
@@ -0,0 +1,42 @@
+      <p>Your account has been created</p>
+    </div>
+    <div class="session-footer">
+      <TmBtn value="Back to Lunie" @click.native="$router.push('/')" />
```
```suggestion
      <TmBtn value="Go to Lunie" @click.native="$router.push('/')" />
```

faboweb commented:
![image](https://user-images.githubusercontent.com/5869273/66984851-da16f900-f0bb-11e9-97fb-03909c84ec69.png)

We should unify the wording. It says seed in the steps and in the confirmation checkbox. @jbibla preferences?

faboweb commented:
The check for the duplicate name is done too late in the process. It should be done immediately on the name step.

faboweb commented:
I think the step points should move a bit more apart. They are very squished together:
![image](https://user-images.githubusercontent.com/5869273/66985038-3c6ff980-f0bc-11e9-92b5-98197707ef3d.png)

jbibla commented:
> I think the step points should move a bit more apart. They are very squished together:

what if we remove step 4 altogether?

mariopino commented:
> The check for the duplicate name is done too late in the process. It should be done immediately on the name step.

Done!
